### DISCUSSION TITLE
Adding Forward Node to Manager node in another subnet

### DISCUSSION THREAD

**Discussion by AhmedMontaaser** [Security-Onion-Solutions/securityonion#7265] [Unsupported Versions]
when i try to add a forward node to manager node but the manager node in subnet 192.168.1.0/24 and the forward node in subnet 192.168.70.0/24 it triggers error that forward node can't accept the salt key remote .. there is a file  created with forward node name under the minions directory .

when im trying to do the same thing but the forward management network card in the same subnet of manager i didn't get that error . 
[notice] the both subnet can connect to each other .
there is the error i got 
![image](https://user-images.githubusercontent.com/97169371/154838018-88ae73a4-8c96-4399-b2cb-5a6f56d8514b.png)

**Comment by dougburks** [comment]
Are there any network firewalls that may be preventing the forward node from fully connecting to the manager?
https://docs.securityonion.net/en/2.3/firewall.html

**Reply by AhmedMontaaser** [reply]
there is a fortigate firewall but the firewall policy is to accept any connection on any port

**Reply by dougburks** [reply]
Have you verified that the forward node can connect to the manager on ports 4505 and 4506 (and the other ports listed at https://docs.securityonion.net/en/2.3/firewall.html) using `nc` or some other tool?

**Reply by AhmedMontaaser** [reply]
yea i did , i checked firewall logs and they can communicate on these ports

**Reply by dougburks** [reply]
Is the firewall doing any inspection of the traffic that could be interfering?

Are there any other devices on your network that could be interfering?

**Reply by AhmedMontaaser** [reply]
i dont think so , here is error file i got 

> Setting up Docker Registry
> Docker Registry Setup - Complete
> Restarting Docker
> Created symlink from /etc/systemd/system/multi-user.target.wants/docker.service to /usr/lib/systemd/system/docker.service.
> ----
> 8% - GENERATING PATCH PILLAR
> ----
> ----
> 8% - INITIALIZING SALT MINION
> ----
> Configuring minion type as sensor
> ----
> 16% - RUNNING FIRST SALT CHECKIN
> ----
> ----
> 20% - ACCEPTING SALT KEY ON MANAGER
> ----
> Accept the key remotely on the manager
> ##########################################
> ##########################################
> ###                                    ###
> ###   UNAUTHORIZED ACCESS PROHIBITED   ###
> ###                                    ###
> ##########################################
> ##########################################
> The key glob 'sensor2_sensor' does not match any unaccepted keys.
> ##########################################
> ##########################################
> ###                                    ###
> ###   UNAUTHORIZED ACCESS PROHIBITED   ###
> ###                                    ###
> ##########################################
> ##########################################
> The key glob 'sensor2_sensor' does not match any unaccepted keys.
> ----
> 21% - COPYING MINION PILLARS TO MANAGER
> ----
> scp pillar and salt files in /root/installtmp to manager /opt/so/saltstack/local
> ##########################################
> ##########################################
> ###                                    ###
> ###   UNAUTHORIZED ACCESS PROHIBITED   ###
> ###                                    ###
> ##########################################
> ##########################################
> ##########################################
> ##########################################
> ###                                    ###
> ###   UNAUTHORIZED ACCESS PROHIBITED   ###
> ###                                    ###
> ##########################################
> ##########################################
> Executing: program /usr/bin/ssh host so, user soremote, command scp -v -r -p -t /tmp/sensor2_sensor/pillar/
> OpenSSH_7.4p1, OpenSSL 1.0.2k-fips  26 Jan 2017
> debug1: Reading configuration data /etc/ssh/ssh_config
> debug1: /etc/ssh/ssh_config line 58: Applying options for *
> debug1: Connecting to so [X.X.X.X] port 22.
> debug1: Connection established.
> debug1: permanently_set_uid: 0/0
> debug1: identity file /root/.ssh/so.key type 1
> debug1: key_load_public: No such file or directory
> debug1: identity file /root/.ssh/so.key-cert type -1
> debug1: Enabling compatibility mode for protocol 2.0
> debug1: Local version string SSH-2.0-OpenSSH_7.4
> debug1: Remote protocol version 2.0, remote software version OpenSSH_7.4
> debug1: match: OpenSSH_7.4 pat OpenSSH* compat 0x04000000
> debug1: Authenticating to so:22 as 'soremote'
> debug1: SSH2_MSG_KEXINIT sent
> debug1: SSH2_MSG_KEXINIT received
> debug1: kex: algorithm: curve25519-sha256
> debug1: kex: host key algorithm: ecdsa-sha2-nistp256
> debug1: kex: server->client cipher: chacha20-poly1305@openssh.com MAC: <implicit> compression: none
> debug1: kex: client->server cipher: chacha20-poly1305@openssh.com MAC: <implicit> compression: none
> debug1: kex: curve25519-sha256 need=64 dh_need=64
> debug1: kex: curve25519-sha256 need=64 dh_need=64
> debug1: expecting SSH2_MSG_KEX_ECDH_REPLY
> debug1: Server host key: ecdsa-sha2-nistp256 SHA256:zCSfF3c7M9Wh/mmncs6Z4KFaAv3jIT1ZZdkextlPAOU
> debug1: Host 'so' is known and matches the ECDSA host key.
> debug1: Found key in /root/.ssh/known_hosts:1
> debug1: rekey after 134217728 blocks
> debug1: SSH2_MSG_NEWKEYS sent
> debug1: expecting SSH2_MSG_NEWKEYS
> debug1: SSH2_MSG_NEWKEYS received
> debug1: rekey after 134217728 blocks
> debug1: SSH2_MSG_EXT_INFO received
> debug1: kex_input_ext_info: server-sig-algs=<rsa-sha2-256,rsa-sha2-512>
> debug1: SSH2_MSG_SERVICE_ACCEPT received
> ##########################################
> ##########################################
> ###                                    ###
> ###   UNAUTHORIZED ACCESS PROHIBITED   ###
> ###                                    ###
> ##########################################
> ##########################################
> debug1: Authentications that can continue: publickey,gssapi-keyex,gssapi-with-mic,password
> debug1: Next authentication method: gssapi-keyex
> debug1: No valid Key exchange context
> debug1: Next authentication method: gssapi-with-mic
> debug1: Unspecified GSS failure.  Minor code may provide more information
> No Kerberos credentials available (default cache: KEYRING:persistent:0)
> 
> debug1: Unspecified GSS failure.  Minor code may provide more information
> No Kerberos credentials available (default cache: KEYRING:persistent:0)
> 
> debug1: Next authentication method: publickey
> debug1: Offering RSA public key: /root/.ssh/so.key
> debug1: Server accepts key: pkalg rsa-sha2-512 blen 279
> debug1: Authentication succeeded (publickey).
> Authenticated to so ([X.X.X.X]:22).
> debug1: channel 0: new [client-session]
> debug1: Requesting no-more-sessions@openssh.com
> debug1: Entering interactive session.
> debug1: pledge: network
> debug1: client_input_global_request: rtype hostkeys-00@openssh.com want_reply 0
> debug1: Sending environment.
> debug1: Sending env LANG = en_US.UTF-8
> debug1: Sending command: scp -v -r -p -t /tmp/sensor2_sensor/pillar/
> File mtime 1611867803 atime 1611867494
> Sending file timestamps: T1611867803 0 1611867494 0
> Sink: T1611867803 0 1611867494 0
> Sending file modes: C0644 300 sensor2_sensor.sls
> Sink: C0644 300 sensor2_sensor.sls
> debug1: client_input_channel_req: channel 0 rtype exit-status reply 0
> debug1: channel 0: free: client-session, nchannels 1
> debug1: fd 0 clearing O_NONBLOCK
> debug1: fd 1 clearing O_NONBLOCK
> debug1: fd 2 clearing O_NONBLOCK
> Transferred: sent 3352, received 3204 bytes, in 0.3 seconds
> Bytes per second: sent 11581.9, received 11070.6
> debug1: Exit status 0
> ##########################################
> ##########################################
> ###                                    ###
> ###   UNAUTHORIZED ACCESS PROHIBITED   ###
> ###                                    ###
> ##########################################
> ##########################################
> Adding sensor2_sensor
> ----
> 22% - CHECKING IF THE SALT MINION NEEDS TO BE UPDATED
> ----
> Attempt to authenticate with the salt master failed with timeout error
> ----
> 23% - GENERATING CA AND CHECKING IN
> ----
> Attempt to authenticate with the salt master failed with timeout error
> Attempt to authenticate with the salt master failed with timeout error
> Attempt to authenticate with the salt master failed with timeout error
> Attempt to authenticate with the salt master failed with timeout error
> Attempt to authenticate with the salt master failed with timeout error
> ----
> 62% - APPLYING FIREWALL SALT STATE
> ----
> Attempt to authenticate with the salt master failed with timeout error
> ----
> 63% - INSTALLING YUM UTILITIES
> ----
> Attempt to authenticate with the salt master failed with timeout error
> ----
>

**Reply by dougburks** [reply]
Please show the full output of the following commands (replacing `manager` with the actual hostname or IP address of your manager):
```
nc -vz manager 4505
nc -vz manager 4506
```

Are you re-running Setup on the forward node with the same hostname of `sensor2`? In general this should work, but you could try with a different hostname just to see if that makes any difference.

**Reply by AhmedMontaaser** [reply]
ok i did it at first 
![image](https://user-images.githubusercontent.com/97169371/159164565-97ceede3-89d6-49cf-b297-ee50d29d2acc.png)
![image](https://user-images.githubusercontent.com/97169371/159164601-dbc788de-3fb0-49a7-8af8-1fa746b6c1af.png)

then i tried to stop the firewalld on manager node and that's result 
it works then it becomes connection refused 

![image](https://user-images.githubusercontent.com/97169371/159165129-a154bcd4-e50b-4862-8994-2764a172f77a.png)

**Reply by AhmedMontaaser** [reply]
netstat output on manager node 
![image](https://user-images.githubusercontent.com/97169371/159175219-dd4fba9d-b41d-4f35-b92d-24ae28908ab4.png)

**Reply by dougburks** [reply]
I asked this in my previous reply but I don't see a response, so I'll repeat it here:
> Are you re-running Setup on the forward node with the same hostname of sensor2? In general this should work, but you could try with a different hostname just to see if that makes any difference.

**Reply by AhmedMontaaser** [reply]
> I asked this in my previous reply but I don't see a response, so I'll repeat it here:
> 
> > Are you re-running Setup on the forward node with the same hostname of sensor2? In general this should work, but you could try with a different hostname just to see if that makes any difference.

ah sorry , i forget to replay this .. yea i changed it and tried another forward node with different name

**Reply by dougburks** [reply]
It seems like something on your network is preventing the forward node from fully communicating with the manager.

**Reply by AhmedMontaaser** [reply]
i tried to simulate the configuration on GNS3 and had the same problem
anyway thanks for your effort

**Comment by its-a-me-mario-the-og** [comment]
Hey I had the same problem and the ports 4505 & 4506 was blocked on the sensor. What we did was 'nc -z -v <ip> 4505 or 4506' to verify it was open or closed. Did the same thing on the manager, if they are closed go into the manger and do 'so-firewall' where you'll get commands to help you. Look at 'so-firewall listhostgroups' and check minion and see if the ip is in the hostgroup. If not add it with 'so-firewall hostgroup minion ip'. Then do 'so-firewall apply' and it will apply the changes then go back into the node and do the nc command and verify if they are open or closed. If they are open redo the install and should be good to go. Hope this helps!
